packages/@react-spectrum/switch/stories/Switch.stories.tsx

```diff
@@ -10,91 +10,104 @@
  * governing permissions and limitations under the License.
  */
 
-import {action} from '@storybook/addon-actions';
-import React from 'react';
-import {storiesOf} from '@storybook/react';
+import {chain} from '@react-aria/utils';
+import {ComponentMeta, ComponentStoryObj} from '@storybook/react';
+import React, {useState} from 'react';
 import {Switch} from '../';
 
-storiesOf('Switch', module)
-  .addParameters({providerSwitcher: {status: 'positive'}})
-  .add(
-    'Default',
-    () => render()
-  )
-  .add(
-    'defaultSelected: true',
-    () => render({defaultSelected: true})
-  )
-  .add(
-    'isSelected: true',
-    () => render({isSelected: true})
-  )
-  .add(
-    'isSelected: false',
-    () => render({isSelected: false})
-  )
-  .add(
-    'isDisabled: true',
-    () => render({isDisabled: true})
-  )
-  .add(
-    'isEmphasized: true',
-    () => render({isEmphasized: true})
-  )
-  .add(
-    'isEmphasized: true, isDisabled: true',
-    () => render({isEmphasized: true, isDisabled: true})
-  )
-  .add(
-    'isReadOnly: true, isSelected: true',
-    () => render({isReadOnly: true, isSelected: true})
-  )
-  .add(
-    'autoFocus: true',
-    () => render({autoFocus: true})
-  )
-  .add(
-    'custom label',
-    () => renderCustomLabel()
-  )
-  .add(
-    'long label',
-    () => (
-      <Switch
-        onChange={action('change')}>
-        Super long checkbox label. Sample text. Arma virumque cano, Troiae qui primus ab oris. Italiam, fato profugus, Laviniaque venit.
-      </Switch>
-    )
-  )
-  .add(
-    'no label',
-    () => renderNoLabel({'aria-label': 'This switch has no visible label'})
-  );
+export default {
+  title: 'Switch',
+  component: Switch,
+  argTypes: {
+    onChange: {
+      action: 'change'
+    },
+    onFocus: {
+      action: 'focus'
+    },
+    onBlur: {
+      action: 'blur'
+    }
+  }
+} as ComponentMeta<typeof Switch>;
 
-function render(props = {}) {
-  return (
-    <Switch
-      onChange={action('change')}
-      {...props}>
-      Switch Label
-    </Switch>
-  );
-}
+export const Default: ComponentStoryObj<typeof Switch> = {
```

Comment:
We might want to do something like this in every file:

`type SwitchStory = ComponentStoryObj<typeof Switch>;`